File: src/ofxconverter/structure/Transaction.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package ofxconverter.structure;

/**
 *
 * @author Floris
 */
public class Transaction {

    private long date = 0;
    private long interestDate = 0;
    private String amount = "820.97";
    private String name = "TEST Transaction name";
    private String account = "122577876";
    private String memo = "TEST Transaction name but with memo's";
    private String type = "OTHER";

    /**
     * @return the date
     */
    public long getDate() {
        return date;
    }

    /**
     * @param date the date to set
     */
    public void setDate(String date) {
        this.date = new Long(date);
    }

    /**
     * @return the amount
     */
    public String getAmount() {
        return amount;
    }

    /**
     * @param amount the amount to set
     */
    public void setAmount(String amount) {
        if( amount.matches( "-?\\d*\\.?\\d+,\\d+$" ) ){
            amount = amount.replace(",", ":");
            amount = amount.replace(".", ",");
            amount = amount.replace(":", ".");
        }
        this.amount = amount;
    }

    /**
     * @return the name
     */
    public String getName() {
        return name;
    }

    /**
     * @param name the name to set
     */
    public void setName(String name) {
        this.name = name;
    }

    /**
     * @return the account
     */
    public String getAccount() {
        return account;
    }

    /**
     * @param account the account to set
     */
    public void setAccount(String account) {
        this.account = account;
    }

    /**
     * @return the memo
     */
    public String getMemo() {
        return memo;
    }

    /**
     * @param memo the memo to set
     */
    public void setMemo(String memo) {
        // Remove all double spaces from the memo field
        memo = memo.replaceAll( "\\s{2,}", " " ).trim();
        this.memo = memo;
    }

    /**
     * @return the type
     */
    public String getType() {
        return type;
    }

    /**
     * @param type the type to set
     */
    public void setType(String type) {
        this.type = type;
    }

    /**
     * @return the interestDate
     */
    public long getInterestDate() {
        return interestDate;
    }

    /**
     * @param interestDate the interestDate to set
     */
    public void setInterestDate(String interestDate) {
        this.interestDate = new Long(interestDate);
    }
}

File: src/ofxconverter/structure/Transaction.py
import re

class Transaction:

    @property
    def interestDate(self):
        return self.__date

    @interestDate.setter
    def interestDate(self, date):
        if date < 0:
            self.__interestDate = 0
        else:
            self.__interestDate = date

    @property
    def date(self):
        return self.__date

    @date.setter
    def date(self, date):
        if date < 0:
            self.__date = 0
        else:
            self.__date = date

    @property
    def amount(self):
        return self.__amount
    
    @amount.setter
    def amount(self, amount):
        if re.search("-?\d*\.?\d+,\d+$", amount):
            amount = amount.replace(",",":")
            amount = amount.replace(".",",")
            amount = amount.replace(":",".")
            self.__amount = amount
        else:
            self.__amount = amount

    @property
    def memo(self):
        return self.__memo
    
    @memo.setter
    def memo(self, memo):
            self.__memo = re.sub("\s{2,}"," ",memo.rstrip()) 

    def __init__(self):
        self.__interestDate = 0
        self.__date = 0
        self.__amount = ""
        self.__memo = ""
        self.name = ""
        self.account = ""
        self.type = "OTHER" 
